html/webappapis/dynamic-markup-insertion/opening-the-input-stream/origin-check-in-document-open.html

```diff
@@ -0,0 +1,36 @@
+<!doctype html>
+<title>Origin check in document.open()</title>
+<link rel="author" title="Jochen Eisinger" href="mailto:jochen@chromium.org">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/#opening-the-input-stream">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/html/resources/common.js"></script>
+<body>
+<script>
+testInIFrame(undefined, (ctx) => {
+  try {
+    ctx.iframes[0].contentDocument.open();
+  } catch (e) {
+    assert_unreached("Opening a same origin document throws");
+  }
+}, "It should be possible to open same origin documents.");
+
+testInIFrame("resources/set-document-domain.html", (ctx) => {
+  try {
+    document.domain = document.domain;
+    ctx.iframes[0].contentDocument.open();
+  } catch (e) {
+    assert_unreached("Opening a same origin document that has its document.domain set throws");
+  }
+}, "It should be possible to open same origin documents when setting document.domain.");
```

Comment:
Maybe add another one here where you open `/common/blank.html` and only set document.domain on this side?

The only problem with document.domain I think is that it's set for the whole document. So once you set it, it's set for all following tests too. So maybe it's better to have separate HTML files for the tests.